### PR TITLE
Updated token.py for gradient calculations

### PR DESCRIPTION
**Pull request by vsamarths**
The `TransformersWordEmbeddings ` forward pass was not put within the `gradient_context`. Hence the forward pass was not encapsulated within the `torch.no_grad()` in case `fine_tuning = False` is set  . This fix improves GPU memory management and speed.

### CODE REVIEW

**Comment by alanakbik** [comment]
@vsamarths thanks for improving this!

@helpmefindaname can you take a look? (Since you're refactoring the TransformerEmbeddings at the moment ;))

**Comment by helpmefindaname** [comment]
This looks good, I've added the change to [my PR ](https://github.com/flairNLP/flair/pull/2558)

**Comment by alanakbik** [comment]
Great! I'll also merge this to master to recognize the contribution! Thanks a lot @vsamarths!
